autopush/web/base.py

```diff
@@ -335,6 +335,14 @@ def _router_fail_err(self, fail, router_type=None, vapid=False, uaid=None):
         # reporting a not found and they're mobile.
         if exc.status_code in [404, 410] and router_type in [
                 'apns', 'fcm', 'adm']:
+            self._base_tags.update({
+                "platform": router_type,
+                "reason": "unregistered",
+            })
```

Comment:
I'm concerned that mutating the base tags here might leak into other usages. Are these tags reused after this point, or are they recreated?

Reply:
It's part of `_router_fail_err` which is an error handler and is one of the last stages for the request Handler.
Mutating the base tags here is fine, and expected.